FlightPlannerD/Storage/FlightStorage.cs

```diff
@@ -0,0 +1,195 @@
+            lock (balanceLock)
+            {
+                foreach (Flight f in _flights.ToList())
+                {
+                    if (f != null)
+                    {
+                        if (f.To != null && f.From != null && f.DepartureTime != null && f.ArrivalTime != null && f.Carrier != null)
+                        {
+                            if ((f.To.AirportCode.Trim().ToUpper() == flight.To.AirportCode.Trim().ToUpper()) &&
+                                 (f.From.AirportCode.Trim().ToUpper() == flight.From.AirportCode.Trim().ToUpper()) &&
+                                 (f.DepartureTime.Trim().ToUpper() == flight.DepartureTime.Trim().ToUpper()) &&
+                                 (f.ArrivalTime.Trim().ToUpper() == flight.ArrivalTime.Trim().ToUpper()) &&
+                                 (f.Carrier.Trim().ToUpper() == flight.Carrier.Trim().ToUpper()))
+                            {
+                                result = true;
+                            }
+                        }
+                    }
+
+                }
+            }
+
+            return result;
+        }
+
+        public static List<Airport> SearchAirport(string part)
+        {
+            part = part.Trim().ToUpper();
+            List<Airport> airport = new List<Airport>();
+            lock (balanceLock)
+            {
+                foreach (var f in _flights)
+                {
+                    if (f.To.AirportCode.Trim().ToUpper().Contains(part) || f.To.City.Trim().ToUpper().Contains(part) || f.To.Country.Trim().ToUpper().Contains(part))
+                    {
+                        if (!airport.Contains(f.To))
+                        {
+                            airport.Add(f.To);
+                        }
+                    }
+                    if (f.From.AirportCode.Trim().ToUpper().Contains(part) || f.From.City.Trim().ToUpper().Contains(part) || f.From.Country.Trim().ToUpper().Contains(part))
+                    {
+                        if (!airport.Contains(f.From))
+                        {
+                            airport.Add(f.From);
+                        }
+                    }
+                }
+            }
+
+            return airport;
+        }
+
+        public static List<Flight> SearchFlights(SearchFlightsRequest flight)
+        {          
+            List<Flight> flights = new List<Flight>();
+            lock (balanceLock)
+            {
+                foreach (var f in _flights)
+                {
+                    if (f.To.AirportCode.Trim().ToUpper() == flight.to.Trim().ToUpper() &&
+                        f.From.AirportCode.Trim().ToUpper() == flight.from.Trim().ToUpper() &&
+                        f.DepartureTime.Contains(flight.departureDate))
+                    {
+                        flights.Add(f);
+                    }
+                }
+            }
+            return flights;
+        }
+
+        public static bool IsFlight(Flight flight)
```

Comment:
IsFlight? Ko tu ar to gribi pārbaudīt?

Reply:
Pārbaudu, vai request  dati pārsūtīti parezii. Edgara analogs bija "IsValid"